File: java/src/main/java/mchehab/com/java/ViewHolder.java
package mchehab.com.java;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

public class ViewHolder extends RecyclerView.ViewHolder {
    private TextView textView;

    public ViewHolder(View itemView) {
        super(itemView);
        textView = itemView.findViewById(R.id.textView);
    }

    public void bindData(Model model){
        textView.setText(model.getText());
    }
}

File: app/src/main/java/mchehab/com/recyclerviewoverview/ViewHolder.kt
package mchehab.com.recyclerviewoverview

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private var textView: TextView = itemView.findViewById(R.id.textView)

    fun bindData(model: Model){
        textView.text = model.text
    }
}
